### ISSUE TITLE
[2.0] Wrong result in tf.keras.losses.BinaryCrossentropy?

### ISSUE THREAD

**Issue by lowkick** [tensorflow/tensorflow#32039]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): YES
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): Binary
- TensorFlow version (use command below): 2.0 beta
- Python version: 3.6
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 10.1 / 7.6.2
- GPU model and memory: rtx 2080 ti (11GB)

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
Trying to translate a model from PyTorch to TF, i've encountered a strange behaviour which I don't understand. The value of the bceloss seems to be different from the value obtained in PyTorch, which instead coincides with the "correct" one.

**Describe the expected behavior**
The expected behaviour is to have the same value.

**Code to reproduce the issue**
out = [0.8800, 0.9271, 0.8596, 0.8748]
target = [1, 1, 1, 1]

expected value: -ln(0.88) - ln(0.9271) - ln(0.8596) - ln(0.8748) = 0.4886/4 = 0.1221

TF: 
tf.keras.losses.BinaryCrossentropy()(out, target)
<tf.Tensor: id=1452660, shape=(), dtype=float32, numpy=1.7575724>

PyTorch:
nn.BCELoss()(out, target)
tensor(0.1221, device='cuda:0', grad_fn=<BinaryCrossEntropyBackward>)

The PyTorch one is the expected value, I don't understand what the TF value is.

**Other info / logs**
Solved temporarily by (re)defining the BCELoss simply copying the source from https://github.com/tensorflow/tensorflow/blob/r2.0/tensorflow/python/keras/backend.py

def real_bce_loss(output, target):
    epsilon = 1e-7
    output = tf.clip_by_value(output, epsilon, 1. - epsilon)
    bce = target * tf.math.log(output + epsilon)
    bce += (1 - target) * tf.math.log(1 - output + epsilon)
    return -tf.reduce_mean(bce)

**Comment by David-Mao** [comment]
I think you should switch the logits and targets.
`tf.keras.losses.BinaryCrossentropy()( [1, 1, 1, 1], [0.8800, 0.9271, 0.8596, 0.8748])` gives you `0.12214369` which is correct.
What you did is to use 1.0 as predicted logits, which causes overflow as it needs `log(0)` in calculation

**Comment by lowkick** [comment]
Easy as that :) Thanks
